x-pack/plugins/observability_solution/infra/server/routes/metadata/lib/get_metric_metadata.ts

```diff
@@ -79,17 +95,25 @@ export const getMetricMetadata = async (
     {
       metrics?: InfraMetadataAggregationResponse;
       nodeName?: InfraMetadataAggregationResponse;
+      monitoredHost?: { name: InfraMetadataAggregationResponse };
     }
   >(requestContext, 'search', metricQuery);
 
   const buckets =
     response.aggregations && response.aggregations.metrics
       ? response.aggregations.metrics.buckets
       : [];
+  const hostWithSystemIntegration =
+    response.aggregations && (response.aggregations?.monitoredHost?.name?.buckets ?? []).length > 0
+      ? response.aggregations?.monitoredHost?.name.buckets[0]?.key
+      : null;
+
+  const hasSystemIntegration = hostWithSystemIntegration === nodeId;
 
   return {
     id: nodeId,
     name: get(response, ['aggregations', 'nodeName', 'buckets', 0, 'key'], nodeId),
+    hasSystemIntegration,
```

Comment:
This attribute kind of makes it harder to extend the same idea to other asset types because this condition is specific to hosts. 

For now, I think it's OK because we don't know if the problem we're solving here will happen to other asset types, but I would return this attribute only when `nodeType === host`. wdyt?

Reply:
I changed that to be returned only for hosts, thanks for catching that!